### PR TITLE
[Partial] Adds API token authentication 

### PR DESCRIPTION
**Pull request by jaenia**
[As a Programmer I write a RESTful endpoint to server the React Frontend](https://trello.com/c/OpftBl6x/26-as-a-programmer-i-write-a-restful-endpoint-to-server-the-react-frontend)

### Description
This PRs authenticates users with a token.

### How to test
- [ ] Go to `http://localhost:8000/api/users/token`
- [ ] Check that if you provide valid credentials, you'll receive an authentication token
- [ ] Check that if you provide invalid credentials, you'll receive a response status of 400 and the message "Unable to authenticate with provided credentials"

### CODE REVIEW

File: backend/api/users/views.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from django.shortcuts import render, redirect  # noqa
+from rest_framework.authtoken.views import ObtainAuthToken
+from rest_framework.settings import api_settings
+
+from api.users.serializers import AuthTokenSerializer
+
+
+class LoginEndpoint(ObtainAuthToken):
```
Comment (filipeximenes) [inline backend/api/users/views.py:8]: It looks like the `ObtainAuthToken` already handles Token creation, you only need to ensure there's a `user` in the `validated_data` of the serializer: https://github.com/encode/django-rest-framework/blob/master/rest_framework/authtoken/views.py#L54

File: backend/api/users/serializers.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            raise serializers.ValidationError(msg, code="authentication")
+
+        attrs["user"] = user
+        return attrs
+
+    def create(self, validated_data):
+        return Token.objects.get_or_create(**validated_data)
+
+    def update(self, instance, validated_data):
+        return instance
```
Comment (filipeximenes) [inline backend/api/users/serializers.py]: You can probably delete these

**Review by filipeximenes — CHANGES_REQUESTED** [review]

File: backend/api/users/views.py (hunk 1 above)
Comment (jaenia) [reply backend/api/users/views.py:8]: Awesome!

File: backend/api/users/serializers.py (hunk 2 above)
Comment (jaenia) [reply backend/api/users/serializers.py]: Good to know :)

**Review by filipeximenes — APPROVED** [review]

**Pull request merged**